packages/components/pie-webc-core/src/decorators/test/required-property.spec.ts

```diff
@@ -0,0 +1,94 @@
+import {
+    beforeEach,
+    afterEach,
+    describe,
+    it,
+    expect,
+    vi,
+} from 'vitest';
+
+import { requiredProperty } from '../required-property';
+
+describe('requiredProperty', () => {
+    let consoleErrorSpy: unknown;
+
+    beforeEach(() => {
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
```

Comment:
For future reference you should be able to use `vi.fn()` instead of an empty arrow function which would avoid the need to disable eslint. Happy for you to try that in your next PR

Reply:
<img width="1279" alt="Screenshot 2023-06-28 at 13 19 37" src="https://github.com/justeattakeaway/pie/assets/16766185/7ed2f95d-778d-4411-a13d-a473888f7df1">

I was hoping so too but I encountered this bad boy when trying, which led me to the sad code you now review

Reply:
You should be able to call mock implet... without anything i.e vi.spyOn(console, 'error').mockImplementation(). Usually that is fine. Or you can specifically return an anonymous method instead.  Or failing that you can just spy on the thing without invoking it that should work too `vi.spyOn(console, 'error')`

Reply:
I think the only problem with `spyOn` without a `mockImplementation` is you still get the error message logging to the console which isn't so nice when you're running the tests locally. 

Your first suggestion would be ideal, however.